### PR TITLE
Fixes #21592 - Add chromedriver as alternative JS debugging

### PR DESCRIPTION
**Pull request by glekner**
Adding chromedriver as a debugging tool for JS integration tests.
Need to add documentation : `JS_TEST_DRIVER=selenium_chrome` for it to run chromedriver

### CODE REVIEW

File: test/integration_test_helper.rb [hunk 1]
```diff
@@ -10,28 +10,40 @@
...
+      :extensions => ["#{Rails.root}/test/integration/support/poltergeist_onload_extensions.js"],
+      :phantomjs => File.join(Rails.root, 'node_modules', '.bin', 'phantomjs')
+    }
+    Capybara::Poltergeist::Driver.new(app, opts)
+  end
 end
 
-Capybara.default_max_wait_time = 30
-Capybara.javascript_driver = :poltergeist
+Capybara.default_max_wait_time = 20
```
Comment (ohadlevy) [inline test/integration_test_helper.rb]: why change the default?

**Comment by mmoll** [comment]
@glekner Could you have a look into GH-5770 to unify the needed driver parts as much as possible (I guess most of it is already in here)?

File: test/integration_test_helper.rb [hunk 2]
```diff
@@ -10,28 +10,41 @@
...
+Capybara.javascript_driver = ENV['JS_TEST_DRIVER'].try(:to_sym) || :poltergeist
```
Comment (ohadlevy) [inline test/integration_test_helper.rb]: you can probably move this inside the if else, so you don't have to check the same condition twice?

File: test/integration_test_helper.rb (hunk 1 above)
Comment (glekner) [reply test/integration_test_helper.rb]: took it from #5770 
https://github.com/theforeman/foreman/pull/5770/files#diff-7a6aaeae51145fd37e71db58212fd18bR43

Comment (ohadlevy) [reply test/integration_test_helper.rb]: IMHO not changing the current behavior will help this PR to merge quicker :)

Comment (glekner) [reply test/integration_test_helper.rb]: k, changed back to 30

**Comment by ohadlevy** [comment]
@bastilian mind having a look? thanks

**Comment by ohadlevy** [comment]
@bastilian mind having a look? thanks

**Comment by ohadlevy** [comment]
@bastilian mind having a look? thanks

**Review by ohadlevy — APPROVED** [review]

**Comment by ohadlevy** [comment]
@ekohl I believe this can be merged as i don't think it needs to be packaged, I did try to look at https://github.com/theforeman/foreman-packaging/blob/rpm/develop/update-requirements#L9 but I didn't see a reference to phantomJS either?

**Review by mmoll — APPROVED** [review]
dev and test deps don't need RPM packaging.

**Comment by ekohl** [comment]
What @mmoll said.

File: test/integration_test_helper.rb [hunk 3]
```diff
@@ -10,28 +10,42 @@
...
-    :phantomjs => File.join(Rails.root, 'node_modules', '.bin', 'phantomjs')
-  }
-  Capybara::Poltergeist::Driver.new(app, opts)
+if ENV["JS_TEST_DRIVER"] == 'selenium_chrome'
+  Selenium::WebDriver::Chrome.driver_path = ENV['TESTDRIVER_PATH'] || File.join(Rails.root, 'node_modules', '.bin', 'chromedriver')
+  Capybara.register_driver :selenium_chrome do |app|
+    options = Selenium::WebDriver::Chrome::Options.new
+    options.args << '--disable-gpu'
+    options.args << '--no-sandbox'
+    options.args << '--window-size=1024,768'
```
Comment (lzap) [inline test/integration_test_helper.rb:28]: Move Over 1024×768: The Most Popular Screen Resolution On The Web Is Now 1366×768 (an article from 7 years ago :-)
https://techcrunch.com/2012/04/11/move-over-1024x768-the-most-popular-screen-resolution-on-the-web-is-now-1366x768/

**Pull request merged**